### PR TITLE
Show locations from incident reports on map

### PR DESCRIPTION
**Pull request by mawaruDOLL**
I updated the event map and event page map to plot locations from incident reports rather than the location collection.

### CODE REVIEW

File: client/controllers/userEvent.coffee [hunk 1]
```diff
@@ -14,6 +14,26 @@ Template.userEvent.onRendered ->
 Template.userEvent.helpers
   isEditing: ->
     return Template.instance().editState.get()
+  incidentLocations: ->
+    locations = {}
+    counts = Template.instance().data.counts
+    # Loop 1: Incident Reports ("counts")
+    for count in counts
+      # Loop 2: Locations within each "count" record
+      for loc in count.locations
```
Comment (sturmer-eha) [inline client/controllers/userEvent.coffee]: There was a bug in this helper, the fixes are available back in the master branch, file locationList.coffee
Namely, there has to be one more line to check if that `count` has any `.locations` has and `date.parse` should be `Date.parse`

Comment (mawaruDOLL) [inline client/controllers/userEvent.coffee]: Ok, I'll introduce the fixes. But just as a side note, this helper won't be staying in the userEvent controller for very long. The map is going to be moved to the locationList template in #43.

File: client/controllers/eventMap.coffee [hunk 2]
```diff
@@ -59,33 +59,44 @@ Template.eventMap.rendered = ->
...
+      filteredEvents = []
       colorScale = chroma.scale(MapHelpers.getDefaultGradientColors()).colors(instance.eventsPerPage)
-
-      while templateEvents.length < instance.eventsPerPage and eventIndex < totalEventCount
-        event = allEvents[eventIndex]
-        eventLocations = instance.data.locations.find({userEventId: event._id}).fetch()
-
-        if eventLocations.length
+      
+      #Remove events that have no locations to plot on the map
```
Comment (sturmer-eha) [inline client/controllers/eventMap.coffee]: It's way easier to read verbose comments when they have a space after the comment symbol.
I found it to be useful to comment code out without a space and put a space and make the first letter capital when comment contains natural language.

**Comment by nathanathan** [comment]
:+1:

**Pull request merged**